version-migration/v0.7-to-v0.8-migration/examples/src/test/kotlin/examples/BreakingChanges.kt

```diff
@@ -0,0 +1,120 @@
+        }
+
+        fun `Example Usage`(): PartiqlAst.Expr.Intersect = PartiqlAst.build {
+            val setQuantifier: PartiqlAst.SetQuantifier = all()
+            val leftBag: PartiqlAst.Expr = bag()
+            val rightBag: PartiqlAst.Expr = bag()
+            intersect(
+                setq = setQuantifier,
+                operands = listOf(leftBag, rightBag),
+                metas = emptyMetaContainer()
+            )
+        }
+    }
+
+    /**
+     * Added: [PassResult]
+     */
+    @Test
+    fun `PassResult (Renamed)`() {
+        fun `Example Usage`(): PassResult<ExprValue> {
+            val ion: IonSystem = IonSystemBuilder.standard().build()
+            val valueFactory: ExprValueFactory = ExprValueFactory.standard(ion)
+            val exprValue: ExprValue = valueFactory.newInt(1)
+            return PassResult.Success(
+                exprValue,
+                emptyList()
+            )
+        }
+    }
+
+    /**
+     * Removed: [ExprValueType.typeNames]
+     */
+    @Test
+    fun `ExprValueType - TypeNames (Removed)`() {
+        fun `Example Usage`() {
+            val nullTypeNames: List<String> = ExprValueType.NULL.typeNames
+            val intTypeNames: List<String> = ExprValueType.INT.typeNames
+        }
+    }
+
+    /**
+     * Changed: Parsing of ORDER BY
+     * Deprecated: [SqlParser]
+     */
+    @Test
+    fun `ORDER BY Parsing (Changed) and Parser (Deprecated)`() {
+
+        /**
+         * The v0.7.* implementation automatically adds a default [PartiqlAst.SortSpec] and a default [PartiqlAst.NullsSpec]
+         * to the parsed [PartiqlAst.OrderBy]. This is removed in v0.8.0.
+         */
+        fun `Example Usage`() {
+            val ion: IonSystem = IonSystemBuilder.standard().build()
+            val customTypes: List<CustomType> = emptyList()
+            val parser: Parser = SqlParser(ion, customTypes)
+            val query = "SELECT * FROM t ORDER BY t.a"
+            val ast: PartiqlAst.Statement.Query = parser.parseAstStatement(query) as PartiqlAst.Statement.Query
+            val exprSelect: PartiqlAst.Expr.Select = ast.expr as PartiqlAst.Expr.Select
+            assert(exprSelect.order?.sortSpecs?.get(0)?.orderingSpec != null)
+            assert(exprSelect.order?.sortSpecs?.get(0)?.nullsSpec != null)
+        }
+    }
+
+    /**
+     * Deprecated: [SqlLexer]
+     *  - Not to be replaced. Unsupported generated files using ANTLR may be used (not as a 1-for-1 replacement) to
+     *    mimic the same functionality -- but PartiQL will not be supporting this use-case. See the `migrated-examples`
+     *    for more information.
+     */
+    @Test
+    fun `SqlLexer (Deprecated)`() {
+        fun `Example Usage`() {
+            val ion: IonSystem = IonSystemBuilder.standard().build()
+            val lexer: Lexer = SqlLexer(ion)
+            lexer.tokenize("SELECT * FROM t")
+        }
+    }
+
+    /**
+     * Deprecated: [TypedOpBehavior.LEGACY]
+     */
```

Comment:
Could add a further comment that the legacy `TypedOpBehavior` is deprecated in v0.8.0 and will later be removed. The default will be `TypedOpBehavior.HONOR_PARAMETERS` as part of v0.9.0 onwards.